modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusStateUpdater.scala

```diff
@@ -172,14 +172,15 @@ object ConsensusStateUpdater {
         case None =>
           for {
             registeredPeers <- consensusStorage.getRegisteredPeers(key)
+            localFacilitators <- (selfId :: registeredPeers).filterA(consensusFns.facilitatorFilter(lastArtifact, _))
             facilitators = facilitatorCalculator.calculate(
               resources.peerDeclarationsMap,
-              selfId :: registeredPeers,
+              localFacilitators,
               resources.removedFacilitators
             )
             time <- Clock[F].monotonic
             effect = consensusStorage.getUpperBound.flatMap { bound =>
-              gossip.spread(ConsensusPeerDeclaration(key, Facility(bound, facilitators.toSet, maybeTrigger)))
+              gossip.spread(ConsensusPeerDeclaration(key, Facility(bound, localFacilitators.toSet, maybeTrigger)))
```

Comment:
Yes, the intention is to spread only locally registered facilitators (after filtering out nodes without collateral). I think it's the correct approach and in the future, it will allow us to implement a "majority superset" (so superset in which peers are added to only if most of the currently added peers are voting for it).